scipy/linalg/basic.py

```diff
@@ -198,7 +198,7 @@ def solve(a, b, sym_pos=False, lower=False, overwrite_a=False,
     if transposed:
         trans = 1
         norm = 'I'
-        if not r_or_c is float:
+        if r_or_c is not float:
             raise ValueError('scipy.linalg.solve can not solve a^T x = b '
                              'for complex matrices. See version 1.1.X.')
```

Comment:
Why is this? Also referring to a version might not be the optimal response.

Reply:
See https://github.com/scipy/scipy/pull/7879#issuecomment-330169984.

Perhaps yes, I should remove explicit mention of version. I will remove once agreed, and the other issue is accepted. ping @pv

Reply:
No I mean why can't we use transpose for complex matrices?

Reply:
Because it is ambiguous whether `a^T x = b` or `a^H x = b` is solved. Some may think that transpose=True is `a^T`, while others `a^H`.

Reply:
@pv, thanks, also the bug yes, `gesvx` accepts 'C' as an argument, not 'H'. So hence lapack failed with wrong argument.

Reply:
Why is it ambigouous? Transpose and transpose conjugate (hermitian) are pretty clear widely used definitions.